changelog/4321.feature.rst

```diff
@@ -0,0 +1,14 @@
+Refactoring of `~sunpy.net.dataretriever` which lead to the following changes:
```

Comment:
This should be split into multiple changelogs

Reply:
Done